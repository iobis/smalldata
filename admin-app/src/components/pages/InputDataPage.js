import React from 'react'
import Divider from '../layout/Divider'
import { useTranslation } from 'react-i18next'
import { Link } from 'react-router-dom'

export default function InputDataPage() {
  const { t } = useTranslation()

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <Link className="button is-info" to="/input-data/new">
            {t('inputDataPage.inputNewOccurrence')}
          </Link>
        </div>
      </section>
      <Divider>{t('inputDataPage.or')}</Divider>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <h4 className="title is-4">{t('inputDataPage.copyPreviousHeader')}</h4>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
              <tr>
                <th/>
                <th>{t('inputDataPage.table.dateAdded')}</th>
                <th>{t('inputDataPage.table.scientificName')}</th>
                <th>{t('inputDataPage.table.dataset')}</th>
                <th>{t('inputDataPage.table.occurrenceDate')}</th>
                <th/>
              </tr>
            </thead>
            <tbody>
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}
