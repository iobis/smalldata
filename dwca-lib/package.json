{
  "name": "@smalldata/dwca-lib",
  "version": "1.0.0",
  "private": true,
  "main": "src/index.js",
  "scripts": {
    "lint": "echo Nothig to lint yet",
    "test:ci": "jest --passWithNoTests"
  },
  "dependencies": {
    "react": "16.8.6"
  },
  "devDependencies": {
    "@babel/core": "7.2.2",
    "@babel/preset-env": "7.3.1",
    "@babel/preset-react": "7.0.0",
    "babel-jest": "24.0.0",
    "jest": "24.7.1",
    "react-test-renderer": "16.8.5"
  }
}
