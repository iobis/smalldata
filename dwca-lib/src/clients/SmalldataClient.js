import { mapDatasetToRequest, mapOccurrenceToDwca } from './SmalldataConverters'

const authorizationValue = 'Basic verysecret'
const headers = {
  'Authorization': authorizationValue,
  'Content-Type':  'application/json'
}

export async function getDatasets() {
  const response = await fetch('/api/datasets', { headers })
    .then(response => response.json())
  return response.map(renameRefToId)
}

export async function createDataset(dataset) {
  const request = mapDatasetToRequest(dataset)
  return await fetch('/api/datasets', {
    method: 'POST',
    headers,
    body:   JSON.stringify(request)
  }).then(response => response.json())
}

export function renameRefToId({ ref, ...rest }) {
  return ({ id: ref, ...rest })
}

export async function getOccurrences({ userRef }) {
  const query = 'projectFields=dwcRecord.tdwg.scientificName&projectFields=dwcRecord.tdwg.eventDate'
  return await fetch(`/api/dwca/user/${userRef}/records?${query}`, { headers })
    .then(response => response.json())
}

export async function getOccurrence({ datasetId, dwcaId, userRef }) {
  return fetch(`/api/dwca/${datasetId}/user/${userRef}/records/${encodeURIComponent(dwcaId)}`, { headers })
    .then(response => response.json())
}

export async function findLatestOccurrence({ userRef }) {
  const occurrences = await getOccurrences({ userRef })
  const latest = occurrences
    .filter(dwca => !!dwca.addedAtInstant)
    .sort((left, right) => left.addedAtInstant < right.addedAtInstant ? 1 : -1)[0]
  return await getOccurrence({ userRef, dwcaId: latest.dwcaId, datasetId: latest.dataset })
}

export async function updateOccurrence({ occurrence, userRef, dwcaId }) {
  const request = mapOccurrenceToDwca(occurrence)
  const url = `/api/dwca/${occurrence.dataset.id}/user/${userRef}/records/${encodeURIComponent(dwcaId)}`
  return await fetch(url, {
    method: 'PUT',
    headers,
    body:   JSON.stringify(request)
  }).then(response => response.json())
}

export async function createOccurrence({ occurrence, userRef }) {
  const request = mapOccurrenceToDwca(occurrence)
  const url = `/api/dwca/${occurrence.dataset.id}/user/${userRef}/records`
  return await fetch(url, {
    method: 'POST',
    headers,
    body:   JSON.stringify(request)
  }).then(response => response.json())
}

export async function getUsers() {
  return fetch('/api/users', { headers })
    .then(response => response.json())
}

export async function getUsersWithDatasets() {
  const [users, datasets] = await Promise.all([getUsers(), getDatasets()])
  const datasetIdToDataset = groupBy(datasets, 'id')
  return users.map(user => {
    if (!user['dataset_refs']) user['dataset_refs'] = []
    const datasets = user['dataset_refs'].map(datasetRef => datasetIdToDataset[datasetRef][0])
    return { ...user, datasets }
  })
}

export async function createUser({ datasetIds, email, name, role }) {
  const request = {
    'dataset_refs': datasetIds,
    emailAddress:   email,
    name,
    role
  }
  const url = '/api/users'
  return await fetch(url, {
    method: 'POST',
    headers,
    body:   JSON.stringify(request)
  }).then(response => response.json())
    .catch(error => ({
      exception:        'Some error occurred',
      exceptionMessage: error
    }))
}

export async function updateUser({ id, datasetIds, email, name, role }) {
  const request = {
    'dataset_refs': datasetIds,
    emailAddress:   email,
    name,
    role
  }
  const url = '/api/users/' + id
  return await fetch(url, {
    method: 'PUT',
    headers,
    body:   JSON.stringify(request)
  }).then(response => response.json())
    .catch(error => ({
      exception:        'Some error occurred',
      exceptionMessage: error
    }))
}

function groupBy(list, props) {
  return list.reduce((a, b) => {
    (a[b[props]] = a[b[props]] || []).push(b)
    return a
  }, {})
}
