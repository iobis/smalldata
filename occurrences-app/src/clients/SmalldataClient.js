import { format } from 'date-fns'

export async function getDatasets() {
  const response = await fetch('/api/datasets')
    .then(response => response.json())
  return response.map(renameRefToId)
}

export function renameRefToId({ ref, ...rest }) {
  return ({ id: ref, ...rest })
}

export function datasetTitleOf(dataset) {
  const title = dataset && dataset.title && dataset.title.value
  return title || ''
}

export function getOccurrenceMock() {
  return [{
    id:             1,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }, {
    id:             2,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }]
}

export async function postOccurrence({ occurence }) {
  const userRef = 'ovZTtaOJZ98xDDY'
  const occurrence = {
    core:       'occurrence',
    occurrence: [{
      tdwg: {
        datasetID:                     occurence.dataset.id,
        datasetName:                   occurence.dataset.title.value,
        basisOfRecord:                 occurence.occurrenceData.basisOfRecord.charAt(0).toUpperCase() + occurence.occurrenceData.basisOfRecord.slice(1),
        eventDate:                     format(occurence.occurrenceData.beginDate, 'YYYY-MM-DD') + '/' + format(occurence.occurrenceData.endDate, 'YYYY-MM-DD'),
        lifestage:                     occurence.occurrenceData.lifeStage,
        occurrenceStatus:              occurence.occurrenceData.occurrenceStatus,
        scientificName:                occurence.occurrenceData.scientificName,
        sex:                           occurence.occurrenceData.sex,
        decimalLongitude:              occurence.locationData.decimalLongitude,
        decimalLatitude:               occurence.locationData.decimalLatitude,
        coordinateUncertaintyInMeters: occurence.locationData.coordinateUncertainty,
        minimumDepthInMeters:          occurence.locationData.minimumDepth,
        maximumDepthInMeters:          occurence.locationData.maximumDepth,
        verbatimCoordinates:           occurence.locationData.verbatimCoordinates,
        verbatimDepth:                 occurence.locationData.verbatimDepth
      }
    }],
    emof:       [{
      purl:  {},
      iobis: {}
    }, {
      iobis: {}
    }]
  }
  const url = `/api/dwca/${occurence.dataset.id}/user/${userRef}/records`
  return await fetch(url, {
    method:  'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body:    JSON.stringify(occurrence)
  }).then(response => response.json())
}
