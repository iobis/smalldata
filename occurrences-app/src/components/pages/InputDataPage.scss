.previous-occurrences {
  margin-top: var(--margin-top);
  .occurrence-row {
    .edit {
      width: 5%;
    }

    .added-at {
      width: 10%;
    }

    .scientific-name {
      width: 10%;
    }

    .dataset-title {
      width: 55%;
    }

    .occurrence-date {
      width: 15%;
    }

    .copy {
      width: 5%;
    }
  }
}

a.button.new-occurrence {
  background-color: var(--obis-green);
  margin-bottom: var(--margin-bottom);

  &:hover {
    background-color: lighten($obis-green, 10%);
  }
}

a.button.is-info.copy-previous-entry {
  background-color: $stepbackground;
}
