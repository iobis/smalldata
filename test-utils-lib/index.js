export function ignoreActWarning() {
  const originalError = console.error

  beforeAll(() => {
    console.error = (...args) => {
      if (/Warning.*not wrapped in act/.test(args[0])) return
      originalError.call(console, ...args)
    }
  })

  afterAll(() => {
    console.error = originalError
  })
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
