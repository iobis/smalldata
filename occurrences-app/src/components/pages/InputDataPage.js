import Divider from '../layout/Divider'
import PropTypes from 'prop-types'
import React, { useContext, useEffect, useState } from 'react'
import { AuthContext } from '@smalldata/dwca-lib'
import { format } from 'date-fns'
import { getDatasets, getOccurrences } from '@smalldata/dwca-lib/src/clients/SmalldataClient'
import { Link } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

export default function InputDataPage() {
  const { t } = useTranslation()
  const { userRef } = useContext(AuthContext)
  const [occurrences, setOccurrences] = useState([])

  useEffect(() => {
    const fetchOccurrences = async() => {
      const [occurrences, datasets] = await Promise.all([
        getOccurrences({ userRef }),
        getDatasets()
      ])
      const datasetRefToTitle = datasets
        .reduce((acc, dataset) => {
          acc[dataset.id] = dataset.title.value
          return acc
        }, {})
      const occurrencesWithDataset = occurrences
        .map(occurrence => ({
          ...occurrence,
          datasetTitle: datasetRefToTitle[occurrence.dataset]
        }))
      setOccurrences(occurrencesWithDataset)
    }
    fetchOccurrences()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <Link className="button is-info new" to="/input-data/create">
            {t('inputDataPage.inputNewOccurrence')}
          </Link>
        </div>
      </section>
      <Divider>{t('inputDataPage.or')}</Divider>
      <section className="previous-occurrences section">
        <div className="container is-fluid has-text-centered">
          <h4 className="title is-4">{t('inputDataPage.copyPreviousHeader')}</h4>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
              <tr>
                <th/>
                <th>{t('inputDataPage.table.addedAt')}</th>
                <th>{t('inputDataPage.table.scientificName')}</th>
                <th>{t('inputDataPage.table.datasetTitle')}</th>
                <th>{t('inputDataPage.table.occurrenceDate')}</th>
                <th/>
              </tr>
            </thead>
            <tbody>
              {occurrences.map(occurrence => (
                <OccurrenceRow
                  datasetId={occurrence.dataset}
                  key={occurrence.dwcaId}
                  occurrenceDate={occurrence.dwcRecords.occurrence[0].tdwg.eventDate}
                  scientificName={occurrence.dwcRecords.occurrence[0].tdwg.scientificName}
                  {...occurrence}/>
              ))}
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}

function OccurrenceRow({ addedAtInstant, datasetId, dwcaId, datasetTitle, occurrenceDate, scientificName }) {
  const { t } = useTranslation()
  const addedAtString = addedAtInstant ? format(addedAtInstant, 'D MMMM YYYY') : '—'
  const toCreate = {
    pathname: '/input-data/create',
    state:    { action: 'create', dwcaId, datasetId }
  }
  const toEdit = {
    pathname: '/input-data/update',
    state:    { action: 'update', dwcaId, datasetId }
  }

  return (
    <tr className="occurrence-row">
      <td className="edit">
        <Link className="button is-info" to={toEdit}>
          {t('common.edit')}
        </Link>
      </td>
      <td className="added-at">{addedAtString}</td>
      <td className="scientific-name">{scientificName}</td>
      <td className="dataset-title">{datasetTitle}</td>
      <td className="occurrence-date">{occurrenceDate || '—'}</td>
      <td className="copy">
        <Link className="button is-info copy-previous-entry" to={toCreate}>
          {t('common.copy')}
        </Link>
      </td>
    </tr>
  )
}

OccurrenceRow.propTypes = {
  addedAtInstant: PropTypes.string,
  datasetId:      PropTypes.string.isRequired,
  datasetTitle:   PropTypes.string.isRequired,
  dwcaId:         PropTypes.string.isRequired,
  occurrenceDate: PropTypes.string,
  scientificName: PropTypes.string.isRequired
}
