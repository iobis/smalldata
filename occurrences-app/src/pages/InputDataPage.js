import React from 'react'
import { getOccurrenceMock } from '../clients/server'
import Divider from '../atom/Divider'
import { useTranslation } from 'react-i18next'

export default function InputDataPage() {
  const { t } = useTranslation()
  const occurrences = getOccurrenceMock()

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <a className="button is-info">{t('inputData.inputNew')}</a>
        </div>
      </section>
      <Divider>{t('inputData.or')}</Divider>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <h3>{t('inputData.copyPreviousHeader')}</h3>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
            <tr>
              <th/>
              <th>{t('inputData.table.dateAdded')}</th>
              <th>{t('inputData.table.scientificName')}</th>
              <th>{t('inputData.table.dataset')}</th>
              <th>{t('inputData.table.occurrenceDate')}</th>
              <th/>
            </tr>
            </thead>
            <tbody>
            {occurrences.map(occurrence => <OccurrenceRow key={occurrence.id} {...occurrence}/>)}
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}

function OccurrenceRow({ id, addedDate, scientificName, dataset, occurrenceDate }) {
  const { t } = useTranslation()

  return (
    <tr>
      <td>
        <div className="button is-info">{t('common.edit')}</div>
      </td>
      <td>{addedDate}</td>
      <td>{scientificName}</td>
      <td>{dataset}</td>
      <td>{occurrenceDate}</td>
      <td>
        <div className="button is-info">{t('common.copy')}</div>
      </td>
    </tr>
  )
}
