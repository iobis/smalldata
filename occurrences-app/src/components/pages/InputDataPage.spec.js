import InputDataPage from './InputDataPage'
import React from 'react'
import { flushPromises, ignoreActWarning } from '@smalldata/test-utils-lib'
import { MemoryRouter } from 'react-router-dom'
import { getDatasetDefaultResponse, OCCURRENCES_RESPONSE } from '@smalldata/dwca-lib/src/clients/SmalldataClient.mock'
import { act } from 'react-dom/test-utils'
import { mount } from 'enzyme'
import { AuthProvider } from '@smalldata/dwca-lib'

describe('InputDataPage', () => {
  ignoreActWarning()

  let wrapper

  beforeEach(() => {
    global.fetch = jest
      .fn()
      .mockImplementationOnce(() =>
        new Promise((resolve) => {
          resolve({ json: () => OCCURRENCES_RESPONSE })
        })
      )
      .mockImplementationOnce(() =>
        new Promise((resolve) => {
          resolve({ json: () => getDatasetDefaultResponse() })
        })
      )
  })

  afterEach(() => {
    global.fetch.mockRestore()
  })

  it('renders correctly', async() => {
    act(() => {
      wrapper = mount(
        <MemoryRouter initialEntries={[{ pathname: '/input-data', key: 'testKey' }]}>
          <AuthProvider>
            <InputDataPage/>
          </AuthProvider>
        </MemoryRouter>
      )
    })

    expect(wrapper.find('tbody tr')).toHaveLength(0)
    expect(wrapper).toMatchSnapshot()
    expect(global.fetch).toHaveBeenCalledTimes(2)
    expect(global.fetch).toHaveBeenNthCalledWith(1, '/api/dwca/user/ovZTtaOJZ98xDDY/records?projectFields=dwcRecord.tdwg.scientificName&projectFields=dwcRecord.tdwg.eventDate', {
      headers: {
        'Content-Type':  'application/json'
      }
    })
    expect(global.fetch).toHaveBeenNthCalledWith(2, '/api/datasets', {
      headers: {
        'Content-Type':  'application/json'
      }
    })

    await flushPromises()
    wrapper.update()
    expect(wrapper.find('tbody tr')).toHaveLength(3)
    expect(wrapper).toMatchSnapshot()
  })
})
