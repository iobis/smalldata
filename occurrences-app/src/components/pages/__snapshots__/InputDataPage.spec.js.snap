// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDataPage renders correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/input-data",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/input-data",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/input-data",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InputDataPage>
      <section
        className="section"
      >
        <div
          className="container is-fluid has-text-centered"
        >
          <Link
            className="button is-info new"
            to="/input-data/create"
          >
            <a
              className="button is-info new"
              href="/input-data/create"
              onClick={[Function]}
            >
              inputDataPage.inputNewOccurrence
            </a>
          </Link>
        </div>
      </section>
      <Divider>
        <div
          className="divider"
        >
          <span>
            inputDataPage.or
          </span>
        </div>
      </Divider>
      <section
        className="previous-occurrences section"
      >
        <div
          className="container is-fluid has-text-centered"
        >
          <h4
            className="title is-4"
          >
            inputDataPage.copyPreviousHeader
          </h4>
          <table
            className="table is-striped is-hoverable is-fullwidth"
          >
            <thead>
              <tr>
                <th />
                <th>
                  inputDataPage.table.addedAt
                </th>
                <th>
                  inputDataPage.table.scientificName
                </th>
                <th>
                  inputDataPage.table.datasetTitle
                </th>
                <th>
                  inputDataPage.table.occurrenceDate
                </th>
                <th />
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </section>
    </InputDataPage>
  </Router>
</MemoryRouter>
`;

exports[`InputDataPage renders correctly 2`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/input-data",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/input-data",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/input-data",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InputDataPage>
      <section
        className="section"
      >
        <div
          className="container is-fluid has-text-centered"
        >
          <Link
            className="button is-info new"
            to="/input-data/create"
          >
            <a
              className="button is-info new"
              href="/input-data/create"
              onClick={[Function]}
            >
              inputDataPage.inputNewOccurrence
            </a>
          </Link>
        </div>
      </section>
      <Divider>
        <div
          className="divider"
        >
          <span>
            inputDataPage.or
          </span>
        </div>
      </Divider>
      <section
        className="previous-occurrences section"
      >
        <div
          className="container is-fluid has-text-centered"
        >
          <h4
            className="title is-4"
          >
            inputDataPage.copyPreviousHeader
          </h4>
          <table
            className="table is-striped is-hoverable is-fullwidth"
          >
            <thead>
              <tr>
                <th />
                <th>
                  inputDataPage.table.addedAt
                </th>
                <th>
                  inputDataPage.table.scientificName
                </th>
                <th>
                  inputDataPage.table.datasetTitle
                </th>
                <th>
                  inputDataPage.table.occurrenceDate
                </th>
                <th />
              </tr>
            </thead>
            <tbody>
              <OccurrenceRow
                addedAtInstant="2019-06-20T09:42:43.788674Z"
                dataset="ntDOtUc7XsRrIus"
                datasetId="ntDOtUc7XsRrIus"
                datasetTitle="Benthos collected in the Azov Sea in 1935 on board the R/V \\"N. Danilevskiy\\""
                dwcRecords={
                  Object {
                    "core": "occurrence",
                    "emof": Array [
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796",
                        "tdwg": Object {},
                      },
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796",
                        "tdwg": Object {},
                      },
                    ],
                    "occurrence": Array [
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796",
                        "tdwg": Object {
                          "eventDate": "1935-10-02",
                          "scientificName": "Nereis succinea",
                        },
                      },
                    ],
                  }
                }
                dwcaId="IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796"
                key="IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796"
                occurrenceDate="1935-10-02"
                scientificName="Nereis succinea"
              >
                <tr
                  className="occurrence-row"
                >
                  <td
                    className="edit"
                  >
                    <Link
                      className="button is-info"
                      to={
                        Object {
                          "pathname": "/input-data/update",
                          "state": Object {
                            "action": "update",
                            "datasetId": "ntDOtUc7XsRrIus",
                            "dwcaId": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info"
                        href="/input-data/update"
                        onClick={[Function]}
                      >
                        common.edit
                      </a>
                    </Link>
                  </td>
                  <td
                    className="added-at"
                  >
                    20 June 2019
                  </td>
                  <td
                    className="scientific-name"
                  >
                    Nereis succinea
                  </td>
                  <td
                    className="dataset-title"
                  >
                    Benthos collected in the Azov Sea in 1935 on board the R/V "N. Danilevskiy"
                  </td>
                  <td
                    className="occurrence-date"
                  >
                    1935-10-02
                  </td>
                  <td
                    className="copy"
                  >
                    <Link
                      className="button is-info copy-previous-entry"
                      to={
                        Object {
                          "pathname": "/input-data/create",
                          "state": Object {
                            "action": "create",
                            "datasetId": "ntDOtUc7XsRrIus",
                            "dwcaId": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_796",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info copy-previous-entry"
                        href="/input-data/create"
                        onClick={[Function]}
                      >
                        common.copy
                      </a>
                    </Link>
                  </td>
                </tr>
              </OccurrenceRow>
              <OccurrenceRow
                addedAtInstant={null}
                dataset="ntDOtUc7XsRrIus"
                datasetId="ntDOtUc7XsRrIus"
                datasetTitle="Benthos collected in the Azov Sea in 1935 on board the R/V \\"N. Danilevskiy\\""
                dwcRecords={
                  Object {
                    "core": "occurrence",
                    "emof": Array [
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158",
                        "tdwg": Object {},
                      },
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158",
                        "tdwg": Object {},
                      },
                    ],
                    "occurrence": Array [
                      Object {
                        "id": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158",
                        "tdwg": Object {
                          "eventDate": "1971",
                          "scientificName": "Brachynotus lucasi",
                        },
                      },
                    ],
                  }
                }
                dwcaId="IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158"
                key="IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158"
                occurrenceDate="1971"
                scientificName="Brachynotus lucasi"
              >
                <tr
                  className="occurrence-row"
                >
                  <td
                    className="edit"
                  >
                    <Link
                      className="button is-info"
                      to={
                        Object {
                          "pathname": "/input-data/update",
                          "state": Object {
                            "action": "update",
                            "datasetId": "ntDOtUc7XsRrIus",
                            "dwcaId": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info"
                        href="/input-data/update"
                        onClick={[Function]}
                      >
                        common.edit
                      </a>
                    </Link>
                  </td>
                  <td
                    className="added-at"
                  >
                    —
                  </td>
                  <td
                    className="scientific-name"
                  >
                    Brachynotus lucasi
                  </td>
                  <td
                    className="dataset-title"
                  >
                    Benthos collected in the Azov Sea in 1935 on board the R/V "N. Danilevskiy"
                  </td>
                  <td
                    className="occurrence-date"
                  >
                    1971
                  </td>
                  <td
                    className="copy"
                  >
                    <Link
                      className="button is-info copy-previous-entry"
                      to={
                        Object {
                          "pathname": "/input-data/create",
                          "state": Object {
                            "action": "create",
                            "datasetId": "ntDOtUc7XsRrIus",
                            "dwcaId": "IBSS_R/V N. Danilevskiy 1935 Azov Sea benthos data_1158",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info copy-previous-entry"
                        href="/input-data/create"
                        onClick={[Function]}
                      >
                        common.copy
                      </a>
                    </Link>
                  </td>
                </tr>
              </OccurrenceRow>
              <OccurrenceRow
                addedAtInstant={null}
                dataset="PoJnGNMaxsupE4w"
                datasetId="PoJnGNMaxsupE4w"
                datasetTitle="Antipatharia distribution data from: Deep-sea fauna of European seas - an annotated species check-list of benthic invertebrates living deeper than 2000 m in the seas bordering Europe"
                dwcRecords={
                  Object {
                    "core": "occurrence",
                    "occurrence": Array [
                      Object {
                        "id": "DeepSea_Antipatharia_35",
                        "tdwg": Object {
                          "scientificName": "Abyssopathes lyra",
                        },
                      },
                    ],
                  }
                }
                dwcaId="DeepSea_Antipatharia_35"
                key="DeepSea_Antipatharia_35"
                scientificName="Abyssopathes lyra"
              >
                <tr
                  className="occurrence-row"
                >
                  <td
                    className="edit"
                  >
                    <Link
                      className="button is-info"
                      to={
                        Object {
                          "pathname": "/input-data/update",
                          "state": Object {
                            "action": "update",
                            "datasetId": "PoJnGNMaxsupE4w",
                            "dwcaId": "DeepSea_Antipatharia_35",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info"
                        href="/input-data/update"
                        onClick={[Function]}
                      >
                        common.edit
                      </a>
                    </Link>
                  </td>
                  <td
                    className="added-at"
                  >
                    —
                  </td>
                  <td
                    className="scientific-name"
                  >
                    Abyssopathes lyra
                  </td>
                  <td
                    className="dataset-title"
                  >
                    Antipatharia distribution data from: Deep-sea fauna of European seas - an annotated species check-list of benthic invertebrates living deeper than 2000 m in the seas bordering Europe
                  </td>
                  <td
                    className="occurrence-date"
                  >
                    —
                  </td>
                  <td
                    className="copy"
                  >
                    <Link
                      className="button is-info copy-previous-entry"
                      to={
                        Object {
                          "pathname": "/input-data/create",
                          "state": Object {
                            "action": "create",
                            "datasetId": "PoJnGNMaxsupE4w",
                            "dwcaId": "DeepSea_Antipatharia_35",
                          },
                        }
                      }
                    >
                      <a
                        className="button is-info copy-previous-entry"
                        href="/input-data/create"
                        onClick={[Function]}
                      >
                        common.copy
                      </a>
                    </Link>
                  </td>
                </tr>
              </OccurrenceRow>
            </tbody>
          </table>
        </div>
      </section>
    </InputDataPage>
  </Router>
</MemoryRouter>
`;
